kernel/vmemitcodes.ml

```diff
@@ -471,7 +578,7 @@ let to_memory code =
   (** Later uses of this string are all purely functional *)
   let code = Bytes.sub_string env.out_buffer 0 env.out_position in
   let code = CString.hcons code in
-  let fold reloc npos accu = (reloc, Array.of_list npos) :: accu in
+  let fold reloc npos accu = (reloc, Positions.of_list (List.rev npos)) :: accu in
```

Comment:
why rev?

Reply:
The Positions code expect the list to be ordered in increasing order, but the relocation code pushes it the other way around.